client/state/sharing/publicize/actions.js

```diff
@@ -113,12 +115,44 @@ export function updateSiteConnection( connection, attributes ) {
 }
 
 /**
+ * Triggers a network request to refresh a Publicize connection for the
+ * specifiedsite ID.
+ *
+ * @param  {Number} siteId           Site ID
+ * @param  {Object} connection       Connection to be deleted.
+ * @param  {String} connection.label Name of the service that was connected.
+ * @return {Function}                Action thunk
+ */
+export function refreshSiteConnection( siteId, connection ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: PUBLICIZE_CONNECTIONS_REQUEST,
+			siteId,
+		} );
+
+		return wpcom.undocumented().siteConnections( siteId )
+			.then( ( response ) => {
+				dispatch( receiveConnections( siteId, response ) );
+				dispatch( {
+					type: PUBLICIZE_CONNECTION_REFRESH,
```

Comment:
> Interesting that we need to duplicate code from fetchConnections to handle the other notice messages.

There is an endpoint to request a single connection but there is no wpcom.undocument wrapper for it. Would creating one be preferable?

> let's rename `PUBLICIZE_CONNECTION_REFRESH` to something like `PUBLICIZE_CONNECTION_REFRESH_SUCCESS` so we don't confuse it with a fetch request.

Should the action names for all four (create/refresh/update/delete) operations be `*_SUCCESS` and `*_FAILURE`? I couldn't really find a consistant naming convention there